libs/parallelism/executors/include/hpx/executors/execution_policy_annotation.hpp

```diff
@@ -0,0 +1,52 @@
+//  Copyright (c) 2021 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file hpx/execution/execution_policy_annotation.hpp
+
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/execution/executors/execution_parameters.hpp>
+#include <hpx/execution/executors/rebind_executor.hpp>
+#include <hpx/execution/traits/is_execution_policy.hpp>
+#include <hpx/properties/property.hpp>
+
+#include <type_traits>
+#include <utility>
+
+namespace hpx { namespace execution { namespace experimental {
+
+    // make_with_annotation property implementation for execution policies
+    // that simply forwards to the embedded executor
+    template <typename ExPolicy,
+        typename Enable =
+            std::enable_if_t<hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::functional::is_tag_invocable<
```

Comment:
Considering @K-ballo's comment: https://github.com/STEllAR-GROUP/hpx/issues/5344#issuecomment-844361465, I should note that I think our `is_tag_invocable` does not currently take into account fallback/override implementations. In this particular case it doesn't matter because `make_with_annotation` has neither a fallback nor an override.

Reply:
Depends what you mean by sufficient. If you're ok with checking those separately, yes. If you expect `is_tag_invocable` on its own to say (in this case) if `make_with_x` is valid, it's not. For that one should currently use just `is_invocable`.

Reply:
In light of #5347 we should leave it as. In order to forcing a sensible compile error I fell back to `tag_fallback`. I agree that this SFINAE here check relies on properties always being implemented in terms of `tag_invoke`.

Reply:
Adding to the previous... Assuming properties are implemented in terms of `tag_invoke` is not an actual limitation as `prefer` uses it to do the dispatching. However what I think we should do is to introduce a trait that allows to check whether invoking a property (i.e. `require` semantics) would succeed compiling. This would allow to separate concerns without having to expose implementation details of `prefer` to user code as currently done above.

Reply:
Good point about `prefer`! Let's keep it like this here and see what we can do separately. Also thanks for #5347, looks good.